validator_test.go

```diff
@@ -0,0 +1,273 @@
+			if tc.expectedErr != nil {
+				assert.ErrorIs(err, tc.expectedErr)
+				return
+			}
+
+			assert.NoError(err)
+		})
+	}
+}
+
+func testNewTypeValidator(t *testing.T) {
+	type Test struct {
+		m                 map[MessageType]Validator
+		defaultValidators Validators
+	}
+
+	tests := []struct {
+		description string
+		value       Test
+		expectedErr error
+	}{
+		// Success case
+		{
+			description: "Default Validators success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Empty map of Validators success",
+			value: Test{
+				m:                 map[MessageType]Validator{},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: nil,
+		},
+		{
+			description: "Omit default Validators success",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{AlwaysValid},
+				},
+			},
+			expectedErr: nil,
+		},
+		// Failure case
+		{
+			description: "Nil list of default Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: AlwaysValid,
+				},
+				defaultValidators: Validators{nil},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Empty list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil Validator error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: nil,
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil list of Validators error",
+			value: Test{
+				m: map[MessageType]Validator{
+					SimpleEventMessageType: Validators{nil},
+				},
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+		{
+			description: "Nil map of Validators error",
+			value: Test{
+				m:                 nil,
+				defaultValidators: Validators{AlwaysValid},
+			},
+			expectedErr: ErrInvalidTypeValidator,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			assert := assert.New(t)
+			msgv, err := NewTypeValidator(tc.value.m, tc.value.defaultValidators...)
+			assert.NotNil(msgv)
+			if tc.expectedErr != nil {
+				assert.ErrorIs(err, tc.expectedErr)
+				return
+			}
+
+			assert.NoError(err)
+		})
+	}
+}
+
+func testAlwaysValid(t *testing.T) {
+	assert := assert.New(t)
+	msg := Message{}
+	err := AlwaysValid(msg)
+	assert.NoError(err)
+}
+
+func testAlwaysInvalid(t *testing.T) {
+	assert := assert.New(t)
+	msg := Message{}
+	err := AlwaysInvalid(msg)
+	assert.ErrorIs(err, ErrInvalidMsgType)
+}
+
+func TestHelperValidators(t *testing.T) {
```

Comment:
What's the benefit of structuring the tests this way vs exporting `testAlwaysInvalid` and `testAlwaysValid`? I don't have an issue with it, just curious.